dateutil/test/test_internals.py

```diff
@@ -12,11 +12,40 @@
 
 import pytest
 
-from dateutil.parser._parser import _ymd
+from dateutil.parser._parser import _ymd, parser
+import dateutil.tz
 
 IS_PY32 = sys.version_info[0:2] == (3, 2)
 
 
+class TestParserPrivate(object):
+    def test_single_character_tzname(self):
+        # See GH#540
+        dstr = '5:50 A.M. on June 13, 1990'
+        res = parser()._parse(dstr)[0]
+        assert res.tzname is None
+
+        dstr = 'Jan 29, 1945 14:45 AM I going to see you there?'
+        res = parser()._parse(dstr, fuzzy=True)[0]
+        assert res.tzname is None
+
+    def test_alpha_tzname(self):
+        # See GH#540
+        dstr = '2017-12-07 10:27:15B'
+        tzinfos = {'B': dateutil.tz.tzoffset('Beta', 120)}
+        res = parser().parse(dstr, tzinfos=tzinfos)
+        assert res.tzinfo is tzinfos['B']
+
+    @pytest.mark.xfail
+    def test_alpha_tzname_ampm(self):
+        # If the Alpha-style timezone is "A" it gets incorrectly
+        # identified as part of AM/PM
+        dstr = '2017-12-07 10:27:15A'
```

Comment:
Hm.. I would guess that `10:27:15A` means AM/PM more often than it means `Alpha`, though I guess they did explicitly set the `tzinfos`...

Reply:
Yah I'm not 100% sure what to do in this case, leaning towards agreeing on the "well if they were explicit...".  Also I think "AM" and "A.M." are way more common than "A" so that should be reasonably safe.